### PR TITLE
Optimizations on the testing backend

### PR DESCRIPTION
**Pull request by fruffy**
- Move IPv6 options out of the critical path
- Move qdisc operations into the kernel target, it is the only target that needs them.
- Improve the test parsing, it is now easier to modify or add options.
- Rename create_filter to compile_dataplane to clarify the purpose of the function

### CODE REVIEW

File: backends/ebpf/run-ebpf-test.py [hunk 1]
```diff
@@ -29,11 +29,44 @@
...
 import tempfile
 import shutil
+import argparse
 sys.path.insert(0, os.path.dirname(
     os.path.realpath(__file__)) + '/../../tools')
 from testutils import *
 
 
+class FullPaths(argparse.Action):
+    """Expand user- and relative-paths"""
```
Comment (mbudiu-vmw) [inline backends/ebpf/run-ebpf-test.py]: Does this need to be a class, or can it be just a function?

**Comment by fruffy** [comment]
Made some improvements. Eventually, I would like to move the run-ebpf-test function out of the ebpf folder and make it a general testing function for all backends. There is a lot of overlap with the bmv2 test framework.
After this pull request has been merged, the xdp backend has to be updated slightly.

**Comment by mbudiu-vmw** [comment]
Are you done with this? Should we merge it?

**Comment by fruffy** [comment]
Yeah can be merged.

**Pull request merged**